kong/cmd/utils/log.lua

```diff
@@ -96,3 +96,26 @@ return setmetatable(_M, {
     return rawget(t, key)
   end
 })
+
+do
+  local pl_utils = require "pl.utils"
+
+  pl_utils.set_deprecation_func(function(msg, trace)
+    if trace then
+      _M.log(ngx.WARN, msg, " ", trace)
+    else
+      _M.log(ngx.WARN, msg)
+    end
+  end)
+
+  _M.deprecatiton = function(message, version_removed, deprecated_after)
+    pl_utils.raise_deprecation({
+      message = message,
+      version_removed = version_removed,
+      deprecated_after = deprecated_after,
+      no_trace = true,
+    })
```

Comment:
I'd leave the trace option in, but inverted, so you can omit it
```suggestion
  _M.deprecatiton = function(message, version_removed, deprecated_after, trace)
    pl_utils.raise_deprecation({
      message = message,
      version_removed = version_removed,
      deprecated_after = deprecated_after,
      no_trace = not trace,
    })
```